backend/django/core/views/api_annotate.py

```diff
@@ -1129,12 +1138,13 @@ def get_label_history(request, project_pk):
     # TODO: annotate uses pk while everything else uses ID. Let's fix this
     data_df["pk"] = data_df["id"]
 
-    # now add back on the metadata fields
-    results = data_df.fillna("").to_dict(orient="records")
-    for i in range(len(results)):
-        results[i]["metadata"] = all_metadata[i]
-        results[i]["formattedMetadata"] = all_metadata_formatted[i]
-        results[i]["metadataIDs"] = page_data_metadata_ids[i]
+    if pre_sorted:
+        page_data = data_df
+    else:
+        df_sorted = data_df.sort_values(by=[sort_by, "data"], key=lambda col: col.str.lower(), ascending=(not reverse))
```

Comment:
Looking into this some more, we have a couple of reasons why this is much slower when sorting by other fields

1. We end up pulling metadata, Labels, IRRLogs for all data instead of the first page of it because the pagination call is moved to the very end.
2. We are using the pandas's sorting method instead of django's

I think the first one is the main culprit though. In a large project with no filters applied, that's going to be a lot of data to pull info for.

One way to speed this up would be to look for code which isn't needed for the new sorting function and move it to after the iloc line. For example: currently you cannot sort by metadata, so you can probably move the metadata calls to only run on the one-page dataset.